File: ViewModels/DictionariesSelectionViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.IO;
using System.Windows.Input;
using Crossword.Infrastructure;
using Crossword.Services.Abstractions;

namespace Crossword.ViewModels;

public class DictionariesSelectionViewModel
{
    public ObservableCollection<DictionaryInfoViewModel> Dictionaries { get; } = new();
    public List<string> SelectionResult { get; } = new();
    public ICommand AcceptCommand { get; }
    public event Action? CloseRequested;

    public DictionariesSelectionViewModel(IDictionaryService dictionaryService)
    {
        var dictionariesPaths = dictionaryService.GetDictionaryPaths();

        foreach (var path in dictionariesPaths)
        {
            var countWords = File.ReadAllLines(path).Length;
            var name = Path.GetFileNameWithoutExtension(path);
            Dictionaries.Add(new DictionaryInfoViewModel(name, countWords));
        }

        AcceptCommand = new RelayCommand(AcceptSelection);
    }

    private void AcceptSelection(object? parameter)
    {
        foreach (var dictInfo in Dictionaries)
        {
            if (!string.IsNullOrEmpty(dictInfo.SelectedWordCount) && int.TryParse(dictInfo.SelectedWordCount, out int count) && count > 0)
            {
                var finalCount = count > dictInfo.TotalWordCount ? dictInfo.TotalWordCount : count;
                SelectionResult.Add($"{dictInfo.Name};{finalCount}");
            }
        }

        CloseRequested?.Invoke();
    }
}

File: crossword_python/viewmodels/DictionariesSelectionViewModel.py
import os
from crossword_python.core.viewmodel_base import ViewModelBase
from crossword_python.core.command import RelayCommand
from crossword_python.viewmodels.DictionaryInfoViewModel import DictionaryInfoViewModel
from PySide6.QtCore import Signal

class DictionariesSelectionViewModel(ViewModelBase):
    close_requested = Signal()

    def __init__(self, dictionary_service):
        super().__init__()
        self.dictionaries: list[DictionaryInfoViewModel] = [ ]
        self.selection_result: list[str] = [ ]
        self.accept_command = RelayCommand(self.accept_selection)

        dict_paths = dictionary_service.get_dictionary_paths()
        for path in dict_paths:
            with open(path, 'r', encoding='utf-8-sig') as f:
                count_words = len(f.readlines())
            name = os.path.splitext(os.path.basename(path))[0]
            self.dictionaries.append(DictionaryInfoViewModel(name, count_words))

    def accept_selection(self, parameter=None):
        self.selection_result.clear()
        for dict_info in self.dictionaries:
            if dict_info.selected_word_count.isdigit():
                count = int(dict_info.selected_word_count)
                if count > 0:
                    final_count = dict_info.total_word_count if count > dict_info.total_word_count else count
                    self.selection_result.append(f"{dict_info.name};{final_count}")
        self.close_requested.emit()